### ISSUE TITLE
JMX exporter reports NaN values for histogram percentile MBeans

### ISSUE THREAD

**Issue by manderson23** [micrometer-metrics/micrometer#331]
I've enabled JMX export in Spring Boot 2.0 M7. I have a histogram named `ingressLatency`. When I selected the `ingressLatency` MBean in JConsole I see the following values.
![screenshot from 2018-01-05 09-20-42](https://user-images.githubusercontent.com/1684641/34603031-1c9b8ae2-f1fa-11e7-86bc-725e9f288a25.png)
However, when I select any of the percentile MBeans (e.g. `ingressLatency.percentile.0.5`) the values are all NaN. I would have expected that they match the values in the screenshot.

**Comment by jkschneider** [comment]
Thanks for the report @manderson23. I've fixed the separate percentile MBeans.

Disclaimer though: Micrometer is using Dropwizard as a passthrough to get metrics to JMX. Dropwizard adds those percentile attributes regardless of whether you've indicated that you want client-side percentiles shipped. The separate `ingressLatency.percentile.X` MBeans are present for any client-side percentiles you've indicated you want in your instrumentation. 

The values will almost never be identical. Dropwizard uses a deeply flawed approach to determining these kind of statistics based on a reservoir interface. The default reservoir in Dropwizard doesn't ever decay old samples (IIRC). The most commonly-used decaying reservoir for Dropwizard _downsamples_ as a memory-saving technique which substantially skews the distribution to lose infrequent but bad latencies (which are the ones you want to see). And the non-downsampling reservoir can consume a ton of memory. So it's a lose-lose-lose proposition.

Micrometer calculates distribution statistics like this off of an internal ring buffer (for decaying old values) of `HdrHistogram`s which maintain a fixed memory footprint and don't do downsampling.

Bottom line: look to the separate percentile MBeans if you care about accuracy. I don't think we'll invest too much in the JMX exposition format as it's kind of more of a one-off diagnostic tool then a full-blown monitoring system.

**Comment by manderson23** [comment]
Thanks for the detailed description. The intention is that JMX is only for dev use and in production we will be using something like Prometheus or Influx.

**Comment by jkschneider** [comment]
Oops, missed the milestone marker. This has been released with 1.0.0-rc.6.

**Comment by izeye** [comment]
I changed the milestone for this as it was marked as "1.0.0-rc.7" somehow but based on @jkschneider 's last comment and Git tags in 1b41f8b93354703c76f8d80314d9705370238c55 this looks belonging to "1.0.0-rc.6".
